app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt

```diff
@@ -320,10 +341,335 @@ class QuestionPlayerActivityTest {
     }
   }
 
+  @Test
+  fun testQuestionPlayer_newHintAndSolVisible_configChange_newHintAndSolVisible() {
+    launchForSkillList(SKILL_ID_LIST_0).use {
+      makeFirstNewHintsVisible(SKILL_ID_0)
```

Comment:
Can you add another assertion in between 
```
     onView(withId(R.id.dot_hint)).check(
        matches(not(isDisplayed()))
      )
```

Reply:
I think we should keep these tests here because these tests are testing the other UI related things when the user views the hints

We can add comments to explicitly mention that  `makeFirstHintVisible` is only for robolectric, or an if condition like 
```
if(isOnRobolectric)
    makeFirstHintVisilible()
```
@anandwana001 @FareesHussain WDYT?

Reply:
I don't think the Robolectric flag works here. As the Behavior is the same across the Test suite.
There can be UI tests for Opening Hints and Solutions and further. but this might be an issue in the future if we add tests for a behavior that is not present in the Actual app.

Let me know If I'm missing somthing here.

Reply:
@MaskedCarrot any update on this?

Reply:
sorry did not see this.
@anandwana001 I agree this can a bit misleading because the hints and solution are already available on espresso( before 60 seconds). But I think we should still add tests here(espresso) to test the UI of HintsAndSolutionDialogFragment.

Reply:
I agree that we need Instrumentation tests for the HintsAndSolutionDialogFragment. but the reason to use the `HintsAndSolutionConfigFastShowTestModule` is not to use espresso for Hints and solutions.

Reply:
I agree that we need instrumentation tests for Hints and solutions, but the reason to use the `HintsAndSolutionConfigFastShowTestModule` is to avoid Hints and solutions tests in espresso.

"we don't want to test them in Espresso since they will likely cause the tests to in the worst-case fail & the best case substantially slow down. Hints & solutions result in Espresso not being idle until they are triggered which makes them hard to test and leads to many of the tests waiting ~30-60 seconds for that state to be hit. We can't reliably test them in Espresso which is why that module was added." -- @BenHenning 

For now the correct way to test is a Robolectric test, Or using UI Automator at the End to End tests in future

Reply:
okay. 
I will update this PR today.